api/app.go

```diff
@@ -99,6 +126,35 @@ func (app *App) Configure() error {
 		return err
 	}
 
+	//we are customizing the default http error reply
+	runtime.HTTPError = func(ctx context.Context, mux *runtime.ServeMux, marshaler runtime.Marshaler,
+		w http.ResponseWriter, _ *http.Request, err error) {
+		w.Header().Set("Content-Type", marshaler.ContentType())
+		var s int
+
+		st, s := app.getStatusCodeFromError(err)
+
+		w.WriteHeader(s)
+
+		type errorBody struct {
+			Success bool   `json:"success"`
+			Reason  string `json:"reason"`
+		}
+
+		body := &errorBody{
+			Success: false,
+			Reason:  st.Message(),
+		}
+
+		buf, merr := marshaler.Marshal(body)
```

Comment:
Suggestion: rename the `err` parameter to `rpcErr` and use good old `err` here; it's the parameter that's unusual, not this one.